oauth2/src/main/java/com/okta/spring/boot/oauth/env/OktaOAuth2PropertiesMappingEnvironmentPostProcessor.java

```diff
@@ -257,13 +289,13 @@ public Object getProperty(@NotNull String key) {
                 // issuer could be null (only resolve properties after checking if the key is `OKTA_OAUTH_ISSUER_WITH_PATH`)
                 return Optional.ofNullable(environment.getProperty(OKTA_OAUTH_ISSUER))
                     .map(issuer -> {
-                        // Check if URL is a Okta Org Authorization Server
-                        if (!issuer.contains("/oauth2")) {
-                            // for the Okta Org Authorization Server, /oauth2 needs to be appended
-                            issuer += "/oauth2";
+                        // Check if URL is an Auth0 org or is an Okta Org Authorization Server
+                        if (isAuth0 || issuer.contains("/oauth2")) {
```

Comment:
@aravindanranganathan-okta why do we do this check? I ignored it for auth0, not sure implications of this

Reply:
Okta's Authorization Server (AS) can be either a) Default AS b) Org/Root AS (Ref https://developer.okta.com/docs/concepts/auth-servers/#available-authorization-server-types). We need to parse the issuer URL accordingly and hence this check.

I am not sure if Auth0 supports both these two types of AS. If so, we should be doing this check. If not, then we can ignore it I think.

Reply:
i'll check if auth0 has this

Reply:
this part is used only for okta domains, for auth0 we get metadata from well-known URL and do not hardcode any values so this check is not needed fo auth0